bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java

```diff
@@ -156,6 +164,10 @@ public Label getLabel() {
 
     @Override
     public @Nullable String getTooltip() {
+        if (xssApi != null) {
+            logger.info("[AF] XSS API has been used to encode tooltip HTML.");
+            return translate(PN_TOOLTIP, xssApi.encodeForHTML(tooltip));
```

Comment:
Although we are saving a rich text through dialog for these props, there seems no existing test case for rich text.
what's the current status for these 2 props? is translation working for rich text?

Reply:
Translation should work. @TalmizAhmed Please enhance the translation collateral /spec which exist today to incorporate this use-case

Reply:
in v2, translation works via key and not string match, hence even with encoding it would work. But fundamentally the PR is backward incompatible, its needs to changed

Reply:
@rismehta Translation does not work with rich text currently
<img width="971" alt="Screenshot 2024-06-11 at 2 47 12 PM" src="https://github.com/adobe/aem-core-forms-components/assets/115221634/d83c8e32-4409-4bb6-b937-e20296dd626e">

Reply:
It works, we already have a test case. You might have missed some steps. Please sync with @ravise5

Reply:
Closing this as this will be unrelated now since the encoding is happening on FormStructParser. Everything else should be working as before without backward incompatibilities.